cmd/status.go

```diff
@@ -0,0 +1,197 @@
+/*
+Copyright 2020 Mirantis, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"runtime"
+	"strconv"
+	"strings"
+
+	"github.com/k0sproject/k0s/pkg/build"
+	"github.com/k0sproject/k0s/pkg/install"
+	ps "github.com/mitchellh/go-ps"
+	"github.com/spf13/cobra"
+	"gopkg.in/yaml.v2"
+)
+
+var (
+	output string
+
+	status    *K0sStatus
+	statusCmd = &cobra.Command{
+		Use:     "status",
+		Short:   "Helper command for get general information about k0s",
+		Example: `The command will return information about system init, PID, k0s role, kubeconfig and similar.`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if runtime.GOOS == "windows" {
+				return fmt.Errorf("currently not supported on windows")
+			}
+
+			var err error
+
+			if status, err = getPid(); err != nil {
+				return err
+			}
+
+			if status.Pid != 0 {
+				if user, err := getProcessOwner(status.Pid); err != nil {
+					return err
+				} else if !strings.Contains(user, "root") {
+					return fmt.Errorf("k0s status should be run as root")
+				}
+
+				if status.SysInit, status.StubFile, err = getSysInit(status.Role); err != nil {
+					return err
+				}
+				if status.Role, err = getRole(status.Pid); err != nil {
+					return err
+				}
+			}
+
+			status.output = output
+			status.String()
+			return nil
+		},
+	}
+)
+
+func init() {
+	status = &K0sStatus{}
+	statusCmd.PersistentFlags().StringVarP(&output, "out", "o", "", "sets type of out put to json or yaml")
+}
+
+type K0sStatus struct {
+	Version  string
+	Pid      int
+	PPid     int
+	Role     string
+	SysInit  string
+	StubFile string
+	output   string
+}
+
+func (s K0sStatus) String() {
+	s.Version = build.Version
```

Comment:
This needs to be the version of the running daemon, not the version of current executable.

Reply:
@jnummelin do you think we can achieve this in any other way?

Reply:
1) write stuff to a file under the run-dir
2) `/proc/$pid/exe version`

Reply:
method 1 would be nice, k0s could pretty much write a `K0sStatus` json to rundir `status.json`